docs/csharp/tour-of-csharp/classes-and-objects.md

```diff
@@ -0,0 +1,725 @@
+
+The ***signature*** of a method must be unique in the class in which the method is declared. The signature of a method consists of the name of the method, the number of type parameters and the number, modifiers, and types of its parameters. The signature of a method does not include the return type.
+
+### Parameters
+
+Parameters are used to pass values or variable references to methods. The parameters of a method get their actual values from the ***arguments*** that are specified when the method is invoked. There are four kinds of parameters: value parameters, reference parameters, output parameters, and parameter arrays.
+
+A ***value parameter*** is used for input parameter passing. A value parameter corresponds to a local variable that gets its initial value from the argument that was passed for the parameter. Modifications to a value parameter do not affect the argument that was passed for the parameter. 
+
+Value parameters can be optional, by specifying a default value so that corresponding arguments can be omitted.
+
+A ***reference parameter*** is used for both input and output parameter passing. The argument passed for a reference parameter must be a variable, and during execution of the method, the reference parameter represents the same storage location as the argument variable. A reference parameter is declared with the ref modifier. The following example shows the use of ref parameters.
+
+```csharp
+using System;
+class RefExample
+{
+	static void Swap(ref int x, ref int y) 
+	{
+		int temp = x;
+		x = y;
+		y = temp;
+	}
+	static void Main() 
+	{
+		int i = 1, j = 2;
+		Swap(ref i, ref j);
+		Console.WriteLine($"{i} {j}");			// Outputs "2 1"
+	}
+}
+```
+
+An ***output parameter*** is used for output parameter passing. An output parameter is similar to a reference parameter except that the initial value of the caller-provided argument is unimportant. An output parameter is declared with the out modifier. The following example shows the use of out parameters.
+
+```csharp
+using System;
+class OutExample
+{
+	static void Divide(int x, int y, out int result, out int remainder) 
+	{
+		result = x / y;
+		remainder = x % y;
+	}
+	static void Main() 
+	{
+		int res, rem;
+		Divide(10, 3, out res, out rem);
+		Console.WriteLine("{0} {1}", res, rem);	// Outputs "3 1"
+	}
+}
+```
+
+A ***parameter array*** permits a variable number of arguments to be passed to a method. A parameter array is declared with the params modifier. Only the last parameter of a method can be a parameter array, and the type of a parameter array must be a single-dimensional array type. The Write and WriteLine methods of the System.Console class are good examples of parameter array usage. They are declared as follows.
```

Comment:
should params be styled as inline code? Also, should we add links to the APIs you're mentioning here too?